---
title: "salfinal"
output: html_document
date: "2024-04-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.



```{r packages and libraries}
library(ggplot2)
library(dplyr)
library(tidyverse)
library(knitr)
library(tidyr)
library(plotly)
library(pals)

```



```{r omni data downloaded fie that have abundance and the one have total AMR number}
#I imported the data tables and merge.

data_omni_s<- read.table(file = "C:/NIST_internship/Sfinal/rm8048-a-omni_1334132_proximeta_report.tsv", header = TRUE, sep = "\t")
#head(data_omni_s)
#this file has same mash reference repeated bec it comes from diff bins with diff completeness


#omni data:
AMR_omni_no<- read.table(file = "C:/NIST_internship/Sfinal/omni_no_amr.tsv", header = TRUE, sep = "\t")
#head(AMR_omni_no)
#this show the no of AMR genes in each bin which may or may not carry the same mash_reference.

#I renamed 'bin_id' in 'AMR_omni_no' to 'cluster_id' to match the other dataframe
AMR_omni_no <- AMR_omni_no %>% 
  rename(cluster_id = bin_id)
#head(AMR_omni_no)

#i added a column of mash_reference so i can plot it in reference to genome
AMR_omni_combined_mash <- merge(AMR_omni_no, data_omni_s[, c("cluster_id", "mash_reference", "abundance", "completeness")], by = "cluster_id")
#head(AMR_omni_combined_mash)

#I Selected the 'mash_reference' and 'abundance' columns to produce a file with those two columns only
in_omni_amr_abundance <- AMR_omni_combined_mash[, c("mash_reference","abundance", "completeness")]

#Summarize relative abundance by mash_reference to
relative_abundance_summary_omni <- in_omni_amr_abundance %>%
  group_by(mash_reference) %>%
  summarise(total_relative_abundance = sum(abundance, na.rm = TRUE))
#write.csv(relative_abundance_summary_omni, "relative_abundance_summary_omni.csv", row.names = FALSE)


#sum(relative_abundance_summary_omni$ total_relative_abundance)
# 33% has AMR genes from omni data

in_omni_amr_total_amr <- AMR_omni_combined_mash[, c("mash_reference","total_amr_genes")]

#Summarize relative total AMR by mash_reference to
in_omni_amr_total_amr_summarize <- in_omni_amr_total_amr %>%
  group_by(mash_reference) %>%
  summarise(total_AMR = sum(total_amr_genes, na.rm = TRUE))
#write.csv(in_omni_amr_total_amr_summarize, "in_omni_amr_total_amr_summarize.csv", row.names = FALSE)

```
 
 
 
 
```{r veg data downloaded fie that have abundance and the one have total AMR number}
#I imported the data tables and merge .

data_veg_s<- read.table(file = "C:/NIST_internship/Sfinal/rm8048-b-veg_1336620_proximeta_report.tsv", header = TRUE, sep = "\t")
#head(data_veg_s)

#omni data:
AMR_veg_no<- read.table(file = "C:/NIST_internship/Sfinal/veg_no_amr.tsv", header = TRUE, sep = "\t")
#head(AMR_veg_no)

#I renamed 'bin_id' in 'AMR_veg_no' to 'cluster_id' to match the other dataframe
AMR_veg_no <- AMR_veg_no %>% 
  rename(cluster_id = bin_id)
#head(AMR_veg_no)

#i added a column of mash_reference so i can plot it in reference to genome
AMR_veg_combined_mash <- merge(AMR_veg_no, data_veg_s[, c("cluster_id", "mash_reference", "abundance", "completeness")], by = "cluster_id")
#head(AMR_veg_combined_mash)

#I Selected the 'mash_reference' and 'abundance' columns to produce a file with those two columns only
in_veg_amr_abundance <- AMR_veg_combined_mash[, c("mash_reference","abundance", "completeness")]


#Summarize relative abundance by mash_reference to
relative_abundance_summary_veg <- in_veg_amr_abundance %>%
  group_by(mash_reference) %>%
  summarise(total_relative_abundance = sum(abundance, na.rm = TRUE))
#write.csv(relative_abundance_summary_veg, "relative_abundance_summary_veg.csv", row.names = FALSE)

#sum(relative_abundance_summary_veg$ total_relative_abundance)
#9% have AMR
# we can compare if there si overlapse in identity

in_veg_amr_total_amr <- AMR_veg_combined_mash[, c("mash_reference","total_amr_genes")]


#Summarize relative total AMR by mash_reference to
in_veg_amr_total_amr_summarize <- in_veg_amr_total_amr %>%
  group_by(mash_reference) %>%
  summarise(total_AMR = sum(total_amr_genes, na.rm = TRUE))
#write.csv(in_veg_amr_total_amr_summarize, "in_veg_amr_total_amr_summarize.csv", row.names = FALSE)




```
 
 

 
 
 
```{r plot Total no of AMR genes in both data}


# Add a source column to each data frame
in_omni_amr_total_amr_summarize <- in_omni_amr_total_amr_summarize %>%
  mutate(Source = 'Omni')
in_veg_amr_total_amr_summarize <- in_veg_amr_total_amr_summarize %>%
  mutate(Source = 'Veg')

# Combine the data from both sources
combined_data_amrno <- bind_rows(in_omni_amr_total_amr_summarize, in_veg_amr_total_amr_summarize)
#write.csv(combined_data_amrno, file = "both_data_AMR.csv", row.names = FALSE)
combined_data_amrno_filter <- combined_data_amrno %>% 
  filter(total_AMR > 2)

# Plot
count_plot_amrno <- ggplot(combined_data_amrno_filter, aes(x=mash_reference, y=total_AMR, fill=Source)) +
  geom_bar(stat="identity", position=position_dodge(), color="black") +
  scale_fill_manual(values = c("Omni" = "green", "Veg" = "blue")) +
  labs(x="Genome", y="Count of AMR Genes", title="Comparison of Number of AMR Genes Across omnivores vs vegeterian diet", fill="Diet Type") +
  theme_minimal() +
  coord_flip()

count_plot_amrno

#s: take abundance greater than specific no or according to completeness, i can start with abundance to top 20 then calculate total abundance, or just show top 20 we need to make it more visual, there is no wrong way try in different ways and pick the best.
#S: u can show this to show all the samples how it looks like and then filter out.


```
 
 
```{r table for AMR number with omly common mash_reference plot}

#I counted the number of sources for each mash_reference
common_refs <- combined_data_amrno %>%
  group_by(mash_reference) %>%
  summarise(source_count = n_distinct(Source)) %>%
  filter(source_count == 2)  #I filtered to keep only those with both sources

# Then, filter the original data frame to keep only the common mash_references
# Create a new data frame for the filtered results
filtered_common_data_amrno <- combined_data_amrno %>%
  filter(mash_reference %in% common_refs$mash_reference)

#print(filtered_common_data_amrno)
kable(filtered_common_data_amrno)
#write.csv(filtered_common_data_amrno, "intersection_microbiome_withAMR.csv", row.names = FALSE)


#plot it 
count_plot_amrno_common <- ggplot(filtered_common_data_amrno, aes(x=mash_reference, y=total_AMR, fill=Source)) +
  geom_bar(stat="identity", position=position_dodge(), color="black") +
  scale_fill_manual(values = c("Omni" = "green", "Veg" = "blue")) +
  labs(x="Genome", y="Count of AMR Genes", title="AMR Genes number comparison  Across omnivores vs vegetarian diet", fill="Diet Type") +
  theme_minimal() +
  coord_flip()

count_plot_amrno_common


```
 
```{r table for AMR total number for the bacteria that present only in omni }

 

#I Counted the number of sources for each mash_reference
source_counts <- combined_data_amrno %>%
  group_by(mash_reference) %>%
  summarise(source_count = n_distinct(Source)) %>%
  filter(source_count == 1)  #I filtered to keep only those with one source

#I filtered to include only mash_references that are exclusively in 'Omni'
omni_exclusive_refs <- combined_data_amrno %>%
  filter(mash_reference %in% source_counts$mash_reference, Source == "Omni")

omni_exclusive_refs_no_source <- omni_exclusive_refs %>%
  select(-Source)


#print(omni_exclusive_refs_no_source)
kable(omni_exclusive_refs_no_source)
#write.csv(omni_exclusive_refs_no_source, "bacteria_only_in_omni_W_AMR.csv", row.names = FALSE)

#2
#produce the same one but with abundance:
source_counts_abundance <- combined_abundance_data %>%
  group_by(mash_reference) %>%
  summarise(source_count = n_distinct(Source)) %>%
  filter(source_count == 1)  #I filtered to keep only those with one source

#I filtered to include only mash_references that are exclusively in 'Omni'
omni_exclusive_refs_abundance <- combined_abundance_data %>%
  filter(mash_reference %in% source_counts$mash_reference, Source == "Omni")

omni_exclusive_refs_no_source_abundance <- omni_exclusive_refs_abundance %>%
  select(-Source)

# Merge the data frames
only_in_omni_new_with_abundance <- merge(omni_exclusive_refs_no_source,
                                         omni_exclusive_refs_no_source_abundance,
                                         by = "mash_reference")

head(only_in_omni_new_with_abundance)
#write.csv(only_in_omni_new_with_abundance, "only_in_omni_new_with_abundance.csv", row.names = FALSE)

```
 


 
 
 
```{r table for AMR total number for the bacteria that present only in veg trial}

 



```
 
 
 
 

 
 
 
 
```{r table for AMR total number for the bacteria that present only in veg}


veg_exclusive_refs <- combined_data_amrno %>%
  filter(mash_reference %in% source_counts$mash_reference, Source == "Veg")

veg_exclusive_refs_no_source <- veg_exclusive_refs %>%
  select(-Source)

#kable(veg_exclusive_refs_no_source)

#write.csv(veg_exclusive_refs_no_source, "bacteria_only_in_veg_W_AMR.csv", row.names = FALSE)


#I filtered to include only mash_references that are exclusively in 'veg'
veg_exclusive_refs_abundance <- combined_abundance_data %>%
  filter(mash_reference %in% source_counts$mash_reference, Source == "Veg")

veg_exclusive_refs_no_source_abundance <- veg_exclusive_refs_abundance %>%
  select(-Source)

# Merge the data frames
only_in_veg_new_with_abundance <- merge(veg_exclusive_refs_no_source,
                                         veg_exclusive_refs_no_source_abundance,
                                         by = "mash_reference")

head(only_in_veg_new_with_abundance)
write.csv(only_in_veg_new_with_abundance, "only_in_veg_new_with_abundance.csv", row.names = FALSE)


```

 
 
```{r abundance table for both diets with a source column}


#I added a source column to each dataframe
relative_abundance_summary_omni <- relative_abundance_summary_omni %>%
  mutate(Source = 'Omni')
relative_abundance_summary_veg <- relative_abundance_summary_veg %>%
  mutate(Source = 'Veg')

# Combined the data all of the data from both sources into one table 
combined_abundance_data <- bind_rows(relative_abundance_summary_omni, relative_abundance_summary_veg)

#print(combined_abundance_data)

#write.csv(combined_abundance_data, "both_data_abundance.csv", row.names = FALSE)

#I counted the number of sources for each mash_reference to identify common ones
common_mash_reference_abundance <- combined_abundance_data %>%
  group_by(mash_reference) %>%
  summarise(source_count = n_distinct(Source)) %>%
  filter(source_count == 2) %>%
  select(mash_reference)  

#I Filtered the original combined data to keep only the common mash_references (common microbiome between 2 diets)
common_abundance_data_intersect <- combined_abundance_data %>%
  filter(mash_reference %in% common_mash_reference_abundance$mash_reference)

#print(common_abundance_data_intersect)

#write.csv(common_abundance_data_intersect, "intersection_microbiome_w_abundance.csv", row.names = FALSE)


```
 
 
 
 
 
```{r plot common_abundance_data_intersect between 2 diets}


# Filter and reorder data based on total_relative_abundance
common_abundance_data_intersect_filtered <- common_abundance_data_intersect %>%
  mutate(mash_reference = fct_reorder(mash_reference, total_relative_abundance))

plot_abundance_intersect_filtered <- ggplot(common_abundance_data_intersect_filtered, aes(x = total_relative_abundance, y = mash_reference, color = Source)) +
  geom_point(alpha = 0.6) +
  scale_color_manual(values = c("Veg" = "blue", "Omni" = "green")) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "right") +
  labs(x = "Relative Abundance", y = "genome", title = "Abundance Distribution by Source",
       color = "Dataset Source")

#I Converted the ggplot object to a plotly object for interactivity
interactive_plot_plot_abundance_intersect_filtered <- ggplotly(plot_abundance_intersect_filtered)

interactive_plot_plot_abundance_intersect_filtered

#S: a set up in the story,use this code: fct reorder by relative abundance

  filter(total_relative_abundance > 0.01) %>%

```
 
 

 
 
 
```{r print the common bacteria in both diets }

#I created separate columns for Omni and Veg total abundance
common_abundance_pivot <- common_abundance_data_intersect %>%
  pivot_wider(names_from = Source, values_from = total_relative_abundance, values_fill = list(total_relative_abundance = 0)) %>%
  rename(Total_Abundance_Omni = Omni, Total_Abundance_Veg = Veg)

#kable(common_abundance_pivot)

#write.csv(common_abundance_pivot, "cbacteria_intersect_abundance.csv", row.names = FALSE)




```
 


```{r Identify core microbiome with significant differences (more than tenfold change) }


# Calculate the ratio and filter significant changes
significant_microbiome_differences <- common_abundance_pivot %>%
  mutate(abundance_ratio = abs(Total_Abundance_Omni / Total_Abundance_Veg)) %>%
  filter(abundance_ratio >= 10 | abundance_ratio <= 0.1) %>%
  arrange(desc(abundance_ratio))  #Iam sorting to see the most significant changes first

kable(significant_microbiome_differences)

#write.csv(significant_microbiome_differences, "intersection_microbiome_differences_10log.csv", row.names = FALSE)



```




```{r Species with the same or similar abundance in both diets}

# Calculate the ratio of abundances and filter for ratios close to 1
similar_abundance_species <- common_abundance_pivot %>%
  mutate(abundance_ratio = Total_Abundance_Omni / Total_Abundance_Veg) %>%
  filter(abundance_ratio >= 0.8, abundance_ratio <= 1.2) %>%
  arrange(desc(abundance_ratio))  # for organism Sorting

kable(similar_abundance_species)

#write.csv(similar_abundance_species, "intersection_similar_abundance_species.csv", row.names = FALSE)



```





```{r summury table for the differences between two diets}

# Q1: Species found in both diets
# Since common_abundance_pivot already represents the intersection, all species in it are found in both diets.
kable(common_abundance_pivot)

# Q2: Species with the same or similar abundance in both diets
kable(similar_abundance_species)

# Q3: Species very high in one diet than the other
kable(significant_microbiome_differences)
# Q4: Similarity in composition of both diets

# Summary table
summary_table_final <- data.frame(
  Description = c("Species in both diets", "Species with similar abundance", "Species significantly different"),
  Count = c(nrow(common_abundance_pivot), nrow(similar_abundance_species), nrow(significant_microbiome_differences))
)

# Display the summary table using kable
kable(summary_table_final)

#S: we can add it in a diagram with inters circles what each diet have and similarities and diff 
# after showing diagrm show the tables
```





```{r}
# Create a data frame to store the log ratio of abundance and sign
div_abd_df <- as.data.frame(matrix(nrow = nrow(common_abundance_pivot), ncol = 3))
colnames(div_abd_df) <- c("mash_reference", "div_abundance", "sign")
div_abd_df$mash_reference <- common_abundance_pivot$mash_reference
div_abd_df$div_abundance <- log10(common_abundance_pivot$Total_Abundance_Veg / common_abundance_pivot$Total_Abundance_Omni)

# Add a new column to indicate the sign (Positive or Negative)
div_abd_df$sign <- ifelse(div_abd_df$div_abundance > 0, "Positive", "Negative")

# Arrange the data for plotting
div_abd_df <- div_abd_df %>%
  mutate(ordering = div_abundance) %>%
  arrange(ordering) %>%
  mutate(mash_reference = factor(mash_reference, levels = unique(mash_reference)))

# Create the plot
abd_div_plot_s <- ggplot(data = div_abd_df, aes(x = mash_reference, y = div_abundance, fill = sign)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_manual(values = c("Positive" = "blue", "Negative" = "red")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.text.y = element_text(colour = "black")) +
  labs(x = "Genome", y = "Log Ratio of Abundance (Veg/Omni)", 
       title = "Log-Scale Abundance Ratios: Veg-Dominant vs. Omni-Dominant Microbiota") +
  coord_flip() +
  ylim(-2, 2)

print(abd_div_plot_s)

#here I am trying to show how much R in each diet 
#I made this graph to see the the diff in abundance between Veggie vs omni diet in the core microbiome where I I divided the #average abundance in the vegetarian group by the average abundance in the omnivorous group, and then taking the base-10 #logarithm of this ratio. This value is stored in the div_abundance column of the div_abd_df data frame.
#the postive means the veg is bigger, the negative means in this bacteria the abundance is bigger in the omni 
#s: don't use the other one is better the interactive.


```


```{r}
correlation <- ggplot(common_abundance_pivot, aes(x = Total_Abundance_Veg, y = Total_Abundance_Omni)) +
  geom_point(aes(color = mash_reference), alpha = 0.6) +  
  geom_smooth(method = "lm", color = "black", se = FALSE) +  
  geom_abline(intercept = 0, slope = 1, linetype = "dashed", color = "red") +  
  scale_x_log10() +  
  scale_y_log10() +  
  labs(x = "Average Abundance in Veg", y = "Average Abundance in Omni", 
       title = "Correlation of Abundance between omni and veggie diet") +
  theme_minimal() +  
  theme(axis.text.x = element_text(angle = 45, hjust = 1),  
        legend.position = "none")

print(correlation)


#This line seems to suggest a positive correlation, indicating that, in general, higher abundances in the Veg diet are associated with higher abundances in the Omni diet.
#The dashed red line is the line of equality (y = x), where points would lie if the average abundance in both diets were the same. Points above this line are more abundant in the Omni diet, while points below this line are more abundant in the Veg diet.
#Most points seem to cluster around the line of equality but below the regression line, suggesting that many microbial species have a similar abundance in both diets.
#Some points are noticeably distant from the line of equality, indicating significant differences in abundance between the diets for these species.
#The regression line is above the line of equality, suggesting that for a given level of abundance in the Veg diet, a slightly higher level might be expected in the Omni diet on 

#S: use as a back slide as postive correlation, i can try abundance over AMR to see if there is a correlation between AMR and abundance from the next tables that have everything. ( u can get back to the table with completeness), this is won;t work because i am dealing with total relative abundance and if I did the same to completness there will no point
```


```{r plot correlation between abundance and completeness for omni data }

completeness_omni <- merge(in_omni_amr_total_amr, in_omni_amr_abundance[, c("mash_reference", "abundance", "completeness")], by ="mash_reference")

completeness_omni_filter <- completeness_omni%>%
  filter(abundance>0.1)


# Plotting the correlation between abundance and total_amr_genes
correlation_omni <- ggplot(completeness_omni_filter, aes(x = completeness , y = completeness)) +
  geom_point(aes(color = total_amr_genes), alpha = 0.6) +  # Points colored by completeness
  geom_smooth(method = "lm", color = "black", se = FALSE) +  # Linear regression line
  scale_color_gradient(low = "blue", high = "red") +  # Color gradient for completeness
  labs(x = "Abundance", y = "Completeness", 
       title = "Correlation of Completeness and Microbial Abundance In Omnivores Diet") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

print(correlation_omni)


```


```{r plot correlation between abundance and completeness for veg data }

completeness_veg <- merge(in_veg_amr_total_amr, in_veg_amr_abundance[, c("mash_reference", "abundance", "completeness")], by ="mash_reference")

completeness_veg_filter <- completeness_veg%>%
  filter(abundance>0.1)


# Plotting the correlation between abundance and total_amr_genes
correlation_veg <- ggplot(completeness_veg_filter, aes(x = completeness , y = completeness)) +
  geom_point(aes(color = total_amr_genes), alpha = 0.6) +  # Points colored by completeness
  geom_smooth(method = "lm", color = "black", se = FALSE) +  # Linear regression line
  scale_color_gradient(low = "blue", high = "red") +  # Color gradient for completeness
  labs(x = "Abundance", y = "Completeness", 
       title = "Correlation of Completeness and Microbial Abundance In Vegeterian Diet") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

print(correlation_veg)


```




```{r scatter plot with mash_reference x-axis, completeness on y-axis and point size representing abundance (omni),color=class}
#
completeness_omni_filter_vs <- completeness_omni%>%
  filter(abundance>0.2)


completeness_vs_abundance_omni<-ggplot(completeness_omni_filter_vs, aes(x = fct_reorder(mash_reference, completeness), y = completeness, size = abundance, color = mash_reference)) +
  geom_point(alpha = 0.6) + 
  scale_size_continuous(range = c(1, 10)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +  
  labs(x = "Genome", y = "Completeness", size = "Abundance", color = "Genome", 
       title = "Abundance vs Completeness For Omnivores diet") +
  theme_minimal() +  
  coord_flip()

completeness_vs_abundance_omni
```



```{r scatter plot with mash_reference x-axis, completeness on y-axis and point size representing abundance (veg),color=class}
#
completeness_veg_filter_vs <- completeness_veg%>%
  filter(abundance>0.2)


completeness_vs_abundance_veg<-ggplot(completeness_veg_filter_vs, aes(x = fct_reorder(mash_reference, completeness), y = completeness, size = abundance, color = mash_reference)) +
  geom_point(alpha = 0.6) + 
  scale_size_continuous(range = c(1, 10)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +  
  labs(x = "Genome", y = "Completeness", size = "Abundance", color = "Genome", 
       title = "Abundance vs Completeness For Vegeterian diet") +
  theme_minimal() +  
  coord_flip()

completeness_vs_abundance_veg
```





```{r}
gqa_log <- ggplot(combined_abundance_data, aes(x = reorder(mash_reference, total_relative_abundance), y = total_relative_abundance, color = Source)) +
  geom_jitter(width = 0.2) +  
  facet_wrap(~Source, ncol = 2) +
  scale_y_log10() +  # Apply log scale to y-axis
  theme(axis.text.x = element_blank()) +
  labs(x = "genome", y = "Abundance (Log Scale)", title = "Comparing Omnivores and Vegeterian Diets Datasets")

plotly_gqa_log <- ggplotly(gqa_log)

plotly_gqa_log

#S: could be used as introduction at begining to show all the data I am working with saying I manipulate it to get more results
```





```{r}
gqa_log <- ggplot(combined_abundance_data, aes(x = reorder(mash_reference, total_relative_abundance), y = total_relative_abundance, color = Source)) +
  geom_jitter(width = 0.2) +  
  #facet_wrap(~Source, ncol = 2) +
  scale_y_log10() +  # Apply log scale to y-axis
  theme(axis.text.x = element_blank()) +
  labs(x = "genome", y = "Abundance (Log Scale)", title = "Comparing Omnivores and Vegeterian Diets Datasets")

plotly_gqa_log <- ggplotly(gqa_log)

plotly_gqa_log

#S: could be used as introduction at begining to show all the data I am working with saying I manipulate it to get m

# plot it this way, anywhere not on the line, means there is a species in both data and it is taking the average < it is hard to draw conc from this we need to further dig, the organisms are diff the two diets look diff from this diagram, if they all unique or have same relative abundance will fall on this sigmoidal curve any far right is higher abundance, the opposite is the low, if they are inline it is present in one sample not the other(unique), any point popping of the line is common 

```


```{r omni data producing file with classes and antimicrobial and total abundance for visualization}
#I am Summarizing data to concatenate unique gene symbols with each cluster_ID

data_omni_pheno_s<- read.table(file = "C:/NIST_internship/Sfinal/omni_full_pheno_table.tsv", header = TRUE, sep = "\t")
#head(data_omni_pheno_s)


summary_data_omni_pheno_s <- data_omni_pheno_s %>%
  group_by(cluster_id) %>%
  summarise(
    unique_genes = toString(unique(Genetic.background)),  # Concatenate unique gene symbols into a single string
    distinct_classes = n_distinct(.data$Class),
    classes_names = toString(unique(.data$Class)),  # Concatenate unique class names into a single string
    distinct_antimicrobials = n_distinct(.data$Antimicrobial),
    antimicrobial_names = toString(unique(.data$Antimicrobial)),  
    .groups = 'drop'  # Drops the grouping structure after summarisation
  )

#write.csv(summary_data_omni_pheno_s, file = "summary_data_omni_pheno_s.csv", row.names = FALSE)
#print(summary_data_omni_pheno_s)


# added file that have mash reference and abundance and total_amr_genes callled AMR_omni_combined_mash to this file

 all_omni_final<- merge(summary_data_omni_pheno_s, AMR_omni_combined_mash[, c("cluster_id","mash_reference", "abundance")], by = "cluster_id")

 
#Summarize relative abundance by mash_reference to
# Summarize with collapsed category details
all_omni_final_collapsed <- all_omni_final %>%
  group_by(mash_reference) %>%
  summarise(
    unique_genes = toString(unique(unique_genes)),
    distinct_classes_number = toString(unique(distinct_classes)),
    classes_names = toString(unique(classes_names)),
    distinct_antimicrobial_number = toString(unique(distinct_antimicrobials)),
    antimicrobial_names = toString(unique(antimicrobial_names)),
    total_relative_abundance = sum(abundance, na.rm = TRUE),
    .groups = 'drop'
  )

#kable(all_omni_final_collapsed)
# Optionally, write to CSV
#write.csv(all_omni_final_collapsed, "all_omni_final_collapsed.csv", row.names = FALSE)

```









```{r veg data producing file with classes and antimicrobial and total abundance for visualization}

#I am Summarizing data to concatenate unique gene symbols with each cluster_ID

data_veg_pheno_s<- read.table(file = "C:/NIST_internship/Sfinal/veg_full_pheno_table.tsv", header = TRUE, sep = "\t")
#head(data_veg_pheno_s)


summary_data_veg_pheno_s <- data_veg_pheno_s %>%
  group_by(cluster_id) %>%
  summarise(
    unique_genes = toString(unique(Genetic.background)),  # Concatenate unique gene symbols into a single string
    distinct_classes = n_distinct(.data$Class),
    classes_names = toString(unique(.data$Class)),  # Concatenate unique class names into a single string
    distinct_antimicrobials = n_distinct(.data$Antimicrobial),
    antimicrobial_names = toString(unique(.data$Antimicrobial)),  
    .groups = 'drop'  # Drops the grouping structure after summarisation
  )

#write.csv(summary_data_veg_pheno_s, file = "summary_data_veg_pheno_s.csv", row.names = FALSE)
#print(summary_data_veg_pheno_s)


# added file that have mash reference and abundance and total_amr_genes callled AMR_veg_combined_mash to this file

 all_veg_final<- merge(summary_data_veg_pheno_s, AMR_veg_combined_mash[, c("cluster_id","mash_reference", "abundance")], by = "cluster_id")

 
#Summarize relative abundance by mash_reference to
# Summarize with collapsed category details
all_veg_final_collapsed <- all_veg_final %>%
  group_by(mash_reference) %>%
  summarise(
    unique_genes = toString(unique(unique_genes)),
    distinct_classes_number = toString(unique(distinct_classes)),
    classes_names = toString(unique(classes_names)),
    distinct_antimicrobial_number = toString(unique(distinct_antimicrobials)),
    antimicrobial_names = toString(unique(antimicrobial_names)),
    total_relative_abundance = sum(abundance, na.rm = TRUE),
    .groups = 'drop'
  )

kable(all_veg_final_collapsed)
# Optionally, write to CSV
#write.csv(all_veg_final_collapsed, "all_veg_final_collapsed.csv", row.names = FALSE)


```


```{r omni counted no antimicrobial Resistance in each mach reference,ploted the mean abundance}



new_try_of_omni <- merge(data_omni_pheno_s, AMR_omni_combined_mash[, c("cluster_id", "mash_reference", "abundance")], by = "cluster_id")

#write.csv(new_try_of_omni, file = "new_try_of_omni.csv", row.names = FALSE)

#plot:

#I Calculated the total counts for each mash_reference
total_counts_newtry_omni <- new_try_of_omni %>%
  group_by(mash_reference) %>%
  summarise(total = n())

#I Calculated the average abundance for each mash_reference
avg_abundance_newtry_omni <- relative_abundance_summary_omni %>%
filter(total_relative_abundance > 0.1)


#I Merged the average abundance with total counts
all_omni_labeled_newtry <- merge(new_try_of_omni, avg_abundance_newtry_omni, by = "mash_reference")
all_omni_labeled_newtry <- merge(all_omni_labeled_newtry, total_counts_newtry_omni, by = "mash_reference")

#I Created the stacked bar chart
omni_org_newtry <- ggplot(all_omni_labeled_newtry, aes(x = reorder(factor(mash_reference), -total), fill = Antimicrobial)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total), size = 3, angle = 0, hjust = -1) +
  labs(x = "Genome", y = "Count of Antimicrobial Resistances in Each Bacteria in Omnivores Diet", title = "Antimicrobial Resistance Profile per Genome in Omnivores Diet") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 20, face = "bold"),  
        axis.title.y = element_text(size = 20, face = "bold"),  
        plot.title = element_text(size = 30, face = "bold", hjust = 0.5)) +  
  
  coord_flip()

omni_org_newtry





```


```{r trying colors}

# the color palette
pal_35 <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
            "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
            "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
            "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5",
            "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C",
            "#F0A0FF", "#0075DC", "#993F00", "#4C005C", "#191919", 
            "#005C31", "#2BCE48", "#FFCC99", "#808080", "#94FFB5")

omni_org_newtry_color <- ggplot(all_omni_labeled_newtry, aes(x = reorder(factor(mash_reference), -total), fill = Antimicrobial)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total), size = 3, angle = 0, hjust = -1) +
  scale_fill_manual(values = pal_35) + 
  labs(x = "Genome", y = "Count of Antimicrobial Resistances in Each Bacteria in Omnivores Diet", 
       title = "Antimicrobial Resistance Profile per Genome in Omnivores Diet") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 20, face = "bold"),  
        axis.title.y = element_text(size = 20, face = "bold"),  
        plot.title = element_text(size = 30, face = "bold", hjust = 0.5)) +  
  coord_flip()

print(omni_org_newtry_color)


```




```{r veg counted number antimicrobial Resistance in each mach reference,ploted the mean abundance}



new_try_of_veg <- merge(data_veg_pheno_s, AMR_veg_combined_mash[, c("cluster_id", "mash_reference", "abundance")], by = "cluster_id")

#plot:

#I Calculated the total counts for each mash_reference
total_counts_newtry_veg <- new_try_of_veg %>%
  group_by(mash_reference) %>%
  summarise(total = n())

#I Calculated the average abundance for each mash_reference
avg_abundance_newtry_veg <- relative_abundance_summary_veg %>%
filter(total_relative_abundance > 0.1)

#I Merged the average abundance with total counts
all_veg_labeled_newtry <- merge(new_try_of_veg, avg_abundance_newtry_veg, by = "mash_reference")
all_veg_labeled_newtry <- merge(all_veg_labeled_newtry, total_counts_newtry_veg, by = "mash_reference")

#I Created the stacked bar chart
veg_org_newtry <- ggplot(all_veg_labeled_newtry, aes(x = reorder(factor(mash_reference), -total), fill = Antimicrobial)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total), size = 3, angle = 0, hjust = -1) +
  labs(x = "Genome", y = "Count of Antimicrobial Resistances in Each Bacteria in Vegetarian Diet", title = "Antimicrobial Resistance Profile per Genome in Vegetarian Diet") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 20, face = "bold"),  
        axis.title.y = element_text(size = 20, face = "bold"),  
        plot.title = element_text(size = 30, face = "bold", hjust = 0.5)) +  
  
  coord_flip()
veg_org_newtry





```


```{r trying colors}

# the color palette
pal_35 <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
            "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
            "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
            "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5",
            "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C",
            "#F0A0FF", "#0075DC", "#993F00", "#4C005C", "#191919", 
            "#005C31", "#2BCE48", "#FFCC99", "#808080", "#94FFB5")

veg_org_newtry_color <- ggplot(all_veg_labeled_newtry, aes(x = reorder(factor(mash_reference), -total), fill = Antimicrobial)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total), size = 3, angle = 0, hjust = -1) +
  scale_fill_manual(values = pal_35) + 
  labs(x = "Genome", y = "Count of Antimicrobial Resistances in Each Bacteria in Vegetarian Diet", 
       title = "Antimicrobial Resistance Profile per Genome in Vegetarian Diet") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 20, face = "bold"),  
        axis.title.y = element_text(size = 20, face = "bold"),  
        plot.title = element_text(size = 30, face = "bold", hjust = 0.5)) +  
  coord_flip()

print(veg_org_newtry_color)


```

```{r count class resistant in omni genome}

unique_mash_by_class_omni <- new_try_of_omni %>%
  distinct(mash_reference, Class, .keep_all = TRUE) %>%
  group_by(mash_reference, Class) %>%
  summarise(
    Average_Abundance = mean(abundance, na.rm = TRUE),
    .groups = 'drop'
  )

#plot:

#I Calculated the total counts for each mash_reference
total_counts_newtry_omni_class <- unique_mash_by_class_omni %>%
  group_by(mash_reference) %>%
  summarise(total = n())

#I Calculated the average abundance for each mash_reference
avg_abundance_newtry_omni_class <- relative_abundance_summary_omni%>%
filter(total_relative_abundance > 0.1)

#I Merged the average abundance with total counts
all_omni_labeled_newtry_class <- merge(unique_mash_by_class_omni, avg_abundance_newtry_omni_class, by = "mash_reference")
all_omni_labeled_newtry_class <- merge(all_omni_labeled_newtry, total_counts_newtry_omni_class, by = "mash_reference")

#I Created the stacked bar chart
omni_org_newtry_class <- ggplot(all_omni_labeled_newtry_class, aes(x = reorder(factor(mash_reference), total.y), fill = Class)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total.y), size = 3, angle = 0, hjust = -1) +
  labs(x = "Genome", y = "Count of Class Resistances in Each Bacteria", title = "Class Resistance Profile per Genome in Omnivores Diet") +
 theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 20, face = "bold"),  
        axis.title.y = element_text(size = 20, face = "bold"),  
        plot.title = element_text(size = 30, face = "bold", hjust = 0.5)) +  
  coord_flip()

omni_org_newtry_class





```


```{r  trial}
library(ggplot2)

omni_org_newtry_class <- ggplot(all_omni_labeled_newtry_class, aes(x = reorder(factor(mash_reference), total_relative_abundance), fill = Class)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total.y), size = 3, angle = 0, hjust = -1) +
  labs(x = "Genome", y = "Count of Class Resistances in Each Bacteria", title = "Class Resistance Profile per Genome in Omnivores Diet") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 14, face = "bold"),  
        axis.title.y = element_text(size = 14, face = "bold"),  
        plot.title = element_text(size = 16, face = "bold")) +  
  coord_flip()

omni_org_newtry_class

```





```{r count class resistant in veg genome}

unique_mash_by_class_veg <- new_try_of_veg %>%
  distinct(mash_reference, Class, .keep_all = TRUE) %>%
  group_by(mash_reference, Class) %>%
  summarise(
    Average_Abundance = mean(abundance, na.rm = TRUE),
    .groups = 'drop'
  )

#plot:

#I Calculated the total counts for each mash_reference
total_counts_newtry_veg_class <- unique_mash_by_class_veg %>%
  group_by(mash_reference) %>%
  summarise(total = n())

#I Calculated the average abundance for each mash_reference
avg_abundance_newtry_veg_class <- relative_abundance_summary_veg%>%
filter(total_relative_abundance > 0.1)

#I Merged the average abundance with total counts
all_veg_labeled_newtry_class <- merge(unique_mash_by_class_veg, avg_abundance_newtry_veg_class, by = "mash_reference")
all_veg_labeled_newtry_class <- merge(all_veg_labeled_newtry, total_counts_newtry_veg_class, by = "mash_reference")

#I Created the stacked bar chart
veg_org_newtry_class <- ggplot(all_veg_labeled_newtry_class, aes(x = reorder(factor(mash_reference), -total.y), fill = Class)) +
  geom_bar(position = "stack", stat = "count") +
  geom_text(aes(label = sprintf("%.2f", total_relative_abundance), y = total.y), size = 3, angle = 0, hjust = -1) +
  labs(x = "Genome", y = "Count of Class Resistances in Each Bacteria", title = "Class Resistance Profile per Genome in Vegetarian Diet") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_text(size = 20, face = "bold"),  
        axis.title.y = element_text(size = 20, face = "bold"),  
        plot.title = element_text(size = 30, face = "bold", hjust = 0.5)) +  
  coord_flip()

veg_org_newtry_class





```



```{r}

```


































